static/js/server_events_dispatch.js

```diff
@@ -48,62 +48,61 @@ exports.dispatch_normal_event = function dispatch_normal_event(event) {
         break;
 
     case 'realm':
-        if (event.op === 'update' && event.property === 'name') {
-            page_params.realm_name = event.value;
-            notifications.redraw_title();
-        } else if (event.op === 'update' && event.property === 'description') {
-            page_params.realm_description = event.value;
-            settings_org.update_realm_description(event.value);
-        } else if (event.op === 'update' && event.property === 'invite_required') {
-            page_params.realm_invite_required = event.value;
-        } else if (event.op === 'update' && event.property === 'invite_by_admins_only') {
-            page_params.realm_invite_by_admins_only = event.value;
-        } else if (event.op === 'update' && event.property === 'inline_image_preview') {
-            page_params.realm_inline_image_preview = event.value;
-        } else if (event.op === 'update' && event.property === 'inline_url_embed_preview') {
-            page_params.realm_inline_url_embed_preview = event.value;
-        } else if (event.op === 'update' && event.property === 'create_stream_by_admins_only') {
-            page_params.realm_create_stream_by_admins_only = event.value;
-            if (!page_params.is_admin) {
-                page_params.can_create_streams = !page_params.realm_create_stream_by_admins_only;
+        var realm_settings = [
+            'add_emoji_by_admins_only',
+            'allow_edit_history',
+            'create_stream_by_admins_only',
+            'default_language',
+            'description',
+            'email_changes_disabled',
+            'inline_image_preview',
+            'inline_url_embed_preview',
+            'invite_by_admins_only',
+            'invite_required',
+            'mandatory_topics',
+            'message_retention_days',
+            'name',
+            'name_changes_disabled',
+            'notifications_stream_id',
+            'restricted_to_domain',
+            'waiting_period_threshold',
+        ];
```

Comment:
I guess maybe that doesn't really help, in that in almost every case there is a function we'll want to call after, so we can't really optimize this list away until we eliminate that.